### PR TITLE
[CHEC-842] Feature/vertical nav

### PR DESCRIPTION
**Pull request by jaepass**
- Added vertical navigation UI component
- I was running into issues with this early on as I was trying to use options as props and wasn't able to nail down how to detect the active item correctly. With suggestions from Robbie and John, refactored to use Vue router's router link component which has its native active class
- Having trouble trying to figure out alternatively how to programmatically push the routes https://router.vuejs.org/guide/essentials/navigation.html#programmatic-navigation in order to add keyboard events (I have not added keyboard events) <- would love some help on this if this is part of a/cs

### CODE REVIEW

File: src/components/ChecVerticalNavItem.vue [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      return this.$route.path;
+    },
+  },
+};
+</script>
+
+<style lang="scss">
+.vertical-navigation {
+  &__item {
+    @apply px-2 py-3 border-gray-200 border-b rounded cursor-pointer;
```
Comment (ScopeyNZ) [inline src/components/ChecVerticalNavItem.vue:58]:
```suggestion
    @apply px-2 py-3 border-gray-200 rounded cursor-pointer text-sm leading-tight;
```

File: src/components/ChecVerticalNavigation.vue [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+<script>
+
+export default {
+  name: 'ChecVerticalNavigation',
+};
+</script>
+
+<style lang="scss">
+.vertical-navigation {
+  @apply rounded shadow-sm w-56 bg-white px-3 py-5;
```
Comment (ScopeyNZ) [inline src/components/ChecVerticalNavigation.vue:23]:
```suggestion
  @apply rounded-md shadow-sm bg-white px-3 py-5;
```

The removal of the `w-56` is something I think we should try to get in the habit of doing. The implementor should be able to choose width I think.

File: src/components/ChecVerticalNavItem.vue [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
```
Comment (ScopeyNZ) [inline src/components/ChecVerticalNavItem.vue:59]:
```suggestion
    & + & {
        @apply border-t;
    }

```
This should get the border between (and not on the last item). Might need to test this before you accept it though.

**Comment by jaepass** [comment]
Thanks Guy will edit stat!

File: src/components/ChecVerticalNavItem.vue (hunk 3 above)
Comment (jaepass) [reply src/components/ChecVerticalNavItem.vue:59]: this worked thanks!

**Pull request merged**